oraclepaas/resource_application_container.go

```diff
@@ -0,0 +1,556 @@
+	log.Printf("[DEBUG] Resource state: %#v", d.State())
+	aClient, err := getApplicationClient(meta)
+	if err != nil {
+		return err
+	}
+	client := aClient.ContainerClient()
+	name := d.Id()
+
+	log.Printf("[DEBUG] Deleting ApplicationClient: %v", name)
+
+	input := application.DeleteApplicationContainerInput{
+		Name: name,
+	}
+	if err := client.DeleteApplicationContainer(&input); err != nil {
+		return fmt.Errorf("Error deleting Application Container: %+v", err)
+	}
+	return nil
+}
+
+func resourceOraclePAASApplicationContainerUpdate(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] Resource state: %#v", d.State())
+
+	log.Print("[DEBUG] Creating application container")
+
+	aClient, err := getApplicationClient(meta)
+	if err != nil {
+		return err
+	}
+	client := aClient.ContainerClient()
+
+	additionalFields := application.UpdateApplicationContainerAdditionalFields{}
+
+	if v, ok := d.GetOk("archive_url"); ok {
+		additionalFields.ArchiveURL = v.(string)
+	}
+
+	if v, ok := d.GetOk("notes"); ok {
+		additionalFields.Notes = v.(string)
+	}
+
+	input := application.UpdateApplicationContainerInput{
+		Name:             d.Get("name").(string),
+		AdditionalFields: additionalFields,
+	}
+
+	if v, ok := d.GetOk("manifest_file"); ok {
+		input.Manifest = v.(string)
+	}
+
+	if v, ok := d.GetOk("manifest_attributes"); ok {
+		manifestAttrs, err := expandManifestAttributes(v.([]interface{})[0].(map[string]interface{}))
+		if err != nil {
+			return err
+		}
+		input.ManifestAttributes = manifestAttrs
+	}
+
+	if v, ok := d.GetOk("deployment_file"); ok {
+		input.Deployment = v.(string)
+	}
+
+	if v, ok := d.GetOk("deployment_attributes"); ok {
+		deploymentAttr, err := expandDeploymentAttributes(d, v.([]interface{})[0].(map[string]interface{}))
+		if err != nil {
+			return err
+		}
+		input.DeploymentAttributes = deploymentAttr
+	}
+
+	info, err := client.UpdateApplicationContainer(&input)
+	if err != nil {
+		return fmt.Errorf("Error updating Application Container: %+v", err)
+	}
+
+	d.SetId(info.Name)
+	return resourceOraclePAASApplicationContainerRead(d, meta)
+}
+
+func expandManifestAttributes(attrs map[string]interface{}) (*application.ManifestAttributes, error) {
+	manifestAttributes := &application.ManifestAttributes{
+		Type:         application.ManifestType(attrs["type"].(string)),
+		StartupTime:  strconv.Itoa(attrs["startup_time"].(int)),
+		ShutdownTime: strconv.Itoa(attrs["shutdown_time"].(int)),
+	}
+	if v := attrs["runtime"]; v != nil {
+		runtimeAttrs := application.Runtime{}
+		runtimeAttrs.MajorVersion = v.([]interface{})[0].(map[string]interface{})["major_version"].(string)
+		manifestAttributes.Runtime = runtimeAttrs
+	}
+	if v := attrs["command"]; v != nil {
+		manifestAttributes.Command = v.(string)
+	}
+	if v := attrs["release"]; v != nil {
+		releaseAttrs := application.Release{}
+		releaseConfig := v.([]interface{})[0].(map[string]interface{})
```

Comment:
Hit a runtime exception here when modifying a config:
```
panic: runtime error: index out of range
2018-07-12T14:19:52.551Z [DEBUG] plugin.terraform-provider-oraclepaas: 
2018-07-12T14:19:52.551Z [DEBUG] plugin.terraform-provider-oraclepaas: goroutine 57 [running]:
2018-07-12T14:19:52.551Z [DEBUG] plugin.terraform-provider-oraclepaas: github.com/terraform-providers/terraform-provider-oraclepaas/oraclepaas.expandManifestAttributes(0xc42040bdd0, 0xb57dc5, 0x13, 0xa2b2a0)
2018-07-12T14:19:52.551Z [DEBUG] plugin.terraform-provider-oraclepaas: 	/vagrant/go/src/github.com/terraform-providers/terraform-provider-oraclepaas/oraclepaas/resource_application_container.go:481 +0x95a
```

Config was:
```
resource "oraclepaas_application_container" "java-app" {
  name               = "EmployeeWebApp"
  archive_url        = "accs-test-apps/employees-web-app.zip"
  runtime            = "java"
  subscription_type  = "HOURLY"

  deployment_attributes {
    memory = "1G"
    instances = 1
  }

  manifest_attributes {
    runtime {
      major_version = "10"
    }
  }
}
```

Output
```
oraclepaas_application_container.java-app: Modifying... (ID: EmployeeWebApp)
  deployment_attributes.0.instances:             "2" => "1"
  manifest_attributes.#:                         "0" => "1"
  manifest_attributes.0.runtime.#:               "0" => "1"
  manifest_attributes.0.runtime.0.major_version: "" => "10"
  manifest_attributes.0.shutdown_time:           "" => "0"
  manifest_attributes.0.startup_time:            "" => "30"
  manifest_attributes.0.type:                    "" => "worker"

Error: Error applying plan:

1 error(s) occurred:
```

Reply:
This is fixed